core/src/main/java/org/apache/calcite/interpreter/AggregateNode.java

```diff
@@ -536,6 +539,29 @@ public MinDouble() {
     }
   }
 
+  /** Implementation of {@code MIN} function to calculate the minimum of
+   * {@code boolean} values as a user-defined aggregate.
+   */
+  public static class MinBoolean {
```

Comment:
For anti join, right input of the join may like this,
    LogicalAggregate(group=[{0}], agg#0=[MIN($1)])
       LogicalProject(X=[$0], $f1=[true])
          LogicalValues(tuples=[[{ 1, 'd' }]])
    so need to calculate the minimum of boolean values, use MinBoolean as a user-defined aggregate.